assets/components/Button.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+export type IPropsVariant = 'primary' | 'secondary' | 'tertiary';
+export type IPropsSize = 'small' | 'medium' | 'large';
+
+export interface IPropsButtonBase {
+    text: string;
+    variant?: IPropsVariant;
+    size?: IPropsSize;
+    id?: string;
+    className?: string;
+    disabled?: boolean;
+    'data-test-id'?: string;
+}
+
+interface IPropsButton extends IPropsButtonBase {
+    type?: 'button',
+    onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
+}
+
+interface IPropsSubmit extends IPropsButtonBase {
+    type: 'submit',
+    onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
```

Comment:
I would still make this mandatory, if we need to put there empty function in case this is not needed it's probably better than forgetting it when it's needed. same goes for reset button